src/invoice_utils/engine/_engine.py

```diff
@@ -4,27 +4,19 @@
 from decimal import Decimal
 from json import JSONDecodeError
 from logging import getLogger
+from typing import Optional
 
 from invoice_utils.models import InvoicedItem
 from invoice_utils.engine._errors import InvoicingInputError, InvoicingInputFormatError
 
 from ._currency import BnrFxRateRule, CurrencyRule
 from ._header import HeaderRule
 
+
 class InvoicingEngine:
-    def __init__(self, file_name: str):
+    def __init__(self, rules: Optional[list[dict]] = None):
```

Comment:
Please make this parameter mandatory (i.e. `list[dict]`) and move setting the default from the initialiser to the calling code.